ath79: TP-Link Archer C7 v4 swap usb led names

howl opened the pull request:
In stock firmare usb1 is upper led and usb2 is lower led.
This change mantain the stock behaviour and makes usb1 port
trigger usb1 led and usb2 port trigger usb2 led.

Signed-off-by: David Santamaría Rogado <howl.nsp@gmail.com>

howl commented:
Matched author with SoB.

oldium commented:
I verified that your change works and swaps the LEDs.

I also checked the stock firmware on my C7 v4 and without your change it behaves the same (lower LED is the USB closer to the blue WAN port). With your change the LEDs are swapped when compared to the stock firmware.

Stock: lower LED is the USB port, which is closer to the blue WAN port
Without your change: lower LED is the USB port, which is closer to the blue WAN port
With your change: upper LED is the USB port, which is closer to the blue WAN port

howl committed:
ath79: TP-Link Archer C7 v4 swap usb led names

In stock firmare usb1 is upper led and usb2 is lower led.
This change mantain the stock behaviour and makes usb1 port
trigger usb1 led and usb2 port trigger usb2 led.

Signed-off-by: David Santamaría Rogado <howl.nsp@gmail.com>

howl commented:
@oldium ok then, I have change it now to follow stock behaviour but assign the leds the same name the port that triggers them. Now the port nearer to the wan port labeled as USB 2 triggers the lower led named usb2.

Thanks for taking time checking it, I now have to check the C7 v2 stock behaviour, as at least in openwrt ar71xx and ath79 the usb nearer to the wan port is also usb2 but triggers the upper led.

howl commented:
I'm not going to have in hand v2 till tomorrow or the next day, so I did some internet research and have found this, https://www.youtube.com/watch?v=jDEB36Su-Jo at 1:50 you can see the upper led is on for the usb2, so it seems it behaves different to v4. It's sure that is v1 or v2 because the wifi interruptor instead of a push button.

chunkeey commented:
Merged. Thanks.

oldium commented:
@howl I have tested your latest change and the result is the following (and I found labels on the back side 😊, so now I found what you were talking about):

* USB port named as _USB2_ on the back side (closer to the blue WAN port) uses lower LED. `dmesg` shows the following when I plug the dongle:

   ```[  657.982699] usb 1-1: new full-speed USB device number 3 using ehci-platform```

   The LED named `tp-link:green:usb2` shines (verified in /sys/class/leds/<name>/brightness).

* USB port named as _USB1_ on the back side (further from the blue WAN port) uses upper LED. `dmesg` shows the following when I plug the dongle:

   ```[  821.781247] usb 2-1: new full-speed USB device number 3 using ehci-platform```

   The LED named `tp-link:green:usb1` shines (verified in /sys/class/leds/<name>/brightness).

Do I understand it correctly that the change is to synchronize the LED name with the label on the back side? Is there a way to synchronize also the USB number shown in `dmesg`?

howl commented:
@oldium no, the usb number you see is defined in the qca956x.dtsi, and follow it's order. That is really not important as it depends in the manufacturer what port is labelled as usb 1 and usb 2, so another device could have labeled usb 1 the usb 1-1 and others like this one usb 2-1. In ar71xx I think usb 1-1 is the usb 1 but there is no device tree to inherit between devices.

The last version of this change makes only a usb leds relabel, to match the name of the led with the usb that triggers it.

I have seen that the original pull request was merged but without the edits since your confirmation that v4 uses lower led for usb2, made a new pull request to mantain the original behavior and match usb port and usb led labels.
